pkgs/development/libraries/freeimage/default.nix

```
@@ -1,48 +1,39 @@
-{ lib, stdenv, fetchurl, unzip, darwin }:
-
-# TODO: consider unvendoring various dependencies (libpng, libjpeg,
-# libwebp, zlib, ...)
+{ lib, stdenv, fetchzip, unzip, darwin, libtiff
+, libpng, zlib, libwebp, libraw, openexr, openjpeg
+, libjpeg, jxrlib, pkgconfig, fetchpatch }:
 
 stdenv.mkDerivation {
-  name = "freeimage-3.18.0";
+  pname = "freeimage";
+  version = "unstable-2020-07-04";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/freeimage/FreeImage3180.zip";
-    sha256 = "1z9qwi9mlq69d5jipr3v2jika2g0kszqdzilggm99nls5xl7j4zl";
+  src = fetchzip {
+    url = "https://sourceforge.net/code-snapshots/svn/f/fr/freeimage/svn/freeimage-svn-r1859.zip";
+    sha256 = "Jl48zEJTyZVfTGWeaWpO31vPOqqHT2xJSmsurMpIJLU=";
   };
+  sourceRoot = "source/FreeImage/trunk";
 
-  patches = lib.optional stdenv.isDarwin ./dylib.patch;
-
-  buildInputs = [ unzip ] ++ lib.optional stdenv.isDarwin darwin.cctools;
-
-  prePatch = if stdenv.isDarwin then ''
-    sed -e 's/$(shell xcrun -find clang)/clang/g' \
-        -e 's/$(shell xcrun -find clang++)/clang++/g' \
-        -e "s|PREFIX = /usr/local|PREFIX = $out|" \
-        -e 's|-Wl,-syslibroot $(MACOSX_SYSROOT)||g' \
-        -e 's|-isysroot $(MACOSX_SYSROOT)||g' \
-        -e 's|	install -d -m 755 -o root -g wheel $(INCDIR) $(INSTALLDIR)||' \
-        -e 's| -m 644 -o root -g wheel||g' \
-        -i ./Makefile.osx
-    # Fix LibJXR performance timers
-    sed 's|^SRCS = \(.*\)$|SRCS = \1 Source/LibJXR/image/sys/perfTimerANSI.c|' -i ./Makefile.srcs
-  '' else ''
-    sed -e s@/usr/@$out/@ \
-        -e 's@-o root -g root@@' \
-        -e 's@ldconfig@echo not running ldconfig@' \
-        -i Makefile.gnu Makefile.fip
-  '';
+  # Ensure that the bundled libraries are not used at all
+  prePatch = "rm -rf Source/Lib* Source/OpenEXR Source/ZLib";
+  patches = [ ./unbundle.diff ];
+
+  nativeBuildInputs = [ unzip pkgconfig ] ++ lib.optional stdenv.isDarwin darwin.cctools;
```

```suggestion
  nativeBuildInputs = [ pkgconfig ] ++ lib.optional stdenv.isDarwin darwin.cctools;
```

`unzip` is not needed if you use `fetchzip` or `fetchsvn` as those already produce directories. It would only be needed by `unpackPhase` to process a tarball that one gets from `fetchurl`.